src/core/config/ip6.h

```diff
@@ -177,4 +177,34 @@
 #define OPENTHREAD_CONFIG_UNSECURE_TRAFFIC_MANAGED_BY_STACK_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_MAX_MULTICAST_SUBSCRIPTION_NUM
+ *
+ * The maximum number of supported IPv6 multicast addresses (with scope >= 4) allows
+ * to be externally added by the device ifself or registered by its child.
+ * At least should be the same as OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS for MTD device.
+ * And allocate more for FTD in case it would proxy additional Multicast address its MTD children
+ * registered.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_MAX_MULTICAST_SUBSCRIPTION_NUM
+#if OPENTHREAD_FTD
```

Comment:
Lots of warnings below.

Reply:
**I believe this is not an issue of this PR.**

With some investigation, I found this is due to missing definitions of `OPENTHREAD_FTD`, `OPENTHREAD_MTD`  and `OPENTHREAD_RADIO` for many Makefile targets (e.g. examples/apps/cli/Makefile.am, examples/apps/ncp/Makefile.am, examples/platforms/simulation/Makefile.am).

A simple fix would be define default values for OPENTHREAD_FTD, OPENTHREAD_MTD and OPENTHREAD_RADIO to be 0. But I think it's better (and expected too?) to define explicit values for each target.